### DISCUSSION TITLE
v6-RC-1 - Release Candidate

### DISCUSSION THREAD

**Discussion by snipe** [snipe/snipe-it#10515] [General]
### ⚠️  IMPORTANT: This version of Snipe-IT REQUIRES PHP 7.4 or greater. 

The long-awaited Release Candidate for Snipe-IT v6 is finally here! There were a lot of things that conspired against us for this to be so late (things like an ongoing global pandemic, Laravel delaying their release because of some unfortunate choices made by the PHP release managers, dependencies being trickier for the same reasons as Laravel was delayed, and so on) but here we are. 

When `develop` and `master` start to diverge so much that it takes *hours* to reconcile merge conflicts, it's in everyone's best interests to push ahead with the release, even if it doesn't have every bell and whistle everyone wanted. We got a *lot* done for v6 - much of it is under the hood and our users won't notice (framework upgrades, security upgrades, etc), but we hope the stuff you *can* see will make it worth it for you. We did end up merging down a LOT of the v6 features (like G Suite Secure LDAP) back down into v5 so that you folks didn't have a wait. **It makes the v6 release a little less dramatic, but we never believe in holding features back just for the sake of publicity.**

As we've stated before, delays in releases for Snipe-IT don't mean the same things as they often do in other software projects, since we were still constantly releasing to the v5.x line while we were developing v6, so it's not as if nothing was moving forward and everyone was holding their breath for v6 just to see some kind of progress. 

And likewise, once v6 is officially released, it doesn't mean we're not going to keep pushing forward as we always do. Like our increasing age, versions are just a number, and all the cool features we had hoped to get into v6 but couldn't will just be moved into the next version. Projects are either feature driven or date driven, they cannot be both. While we are normally feature-driven, that's because of our very fast release cycle. 

We've been working on this for a long time (since about June 2021 when we broke the `develop` branch off from the normal v5 bloodline). We hope you'll consider testing out the RC versions so we can make sure the rollout is as smooth as possible. **As ALWAYS, do not use release candidates in production.**

**Note:** For consistency purposes, we'll be carrying this official v5 to v6 changelog below forward into the final release of v6.0.0 to save me the time of having to pull out save months worth of changes for each RC. :) 

### Added 
- Added PHP8 compatibility 
- Restore from previous Snipe-IT backup added to web UI
- Added ability to upload asset images via API
- Add components and users count to dashboard
- Added dashboard module for locations [ch9199]  
- Added asset restore to API
- Added reqestable as model bulk edit field 
- Added stricter validation for slack hooks (people were trying to use non-slack urls for webhooks)
- Added additional fields to new user modal [#10421]
- Adding `.my.cnf` to disable `column-statistics` backup for Docker

### Improved
- Tons of mobile and web UX/UI improvements [#10129]
- Optimized queries for faster loading 
- Upgraded from Laravel 6 to Laravel 8
- Improved Remote User login: support Google Cloud IAP headers
- Improved S3 compatibility 
- Upgraded to latest and created Font Awesome
- Exclude web.config from Apache
- Many improvements to localization, adding more translatable strings to the UI

### Fixed
- TONS of bugfixes (too many to list)
- Improved Docker support
- Improved accessibility
- Reverted back to the old LDAP login and sync methods from AdLdap, which had caused more problems than it solved in the long run
- Fixed #8828: Can't search by checked out user in advanced search  
- Fixed extraneous wildcard query in asset search [#10514]

### Removed
- Dropped support for PHP 7.3 (PHP 7.3 [is already EOL](https://www.php.net/supported-versions.php), so you should have definitely upgraded anyway)

For a full list of changes, [see the complete changelog](https://github.com/snipe/snipe-it/compare/v5.3.7...v6.0.0-RC-1). To see the details on the 260 merged PRs to `develop` since the release of v5, click [here](https://github.com/snipe/snipe-it/pulls?page=10&q=is%3Apr+base%3Adevelop+is%3Amerged+merged%3A%3E2020-10-20).

### Upgrading

For general upgrading instructions, [click here](https://snipe-it.readme.io/docs/upgrading). Users who installed Snipe-IT via Git (recommended) can just run `php upgrade.php`.

<hr /><em>This discussion was created from the release <a href='https://github.com/snipe/snipe-it/releases/tag/v6.0.0-RC-1'>v6-RC-1 - Release Candidate</a>.</em>

**Comment by matmair** [comment]
@snipe thanks for everything you and your team are doing here. Good stuff!

**Comment by iamkhris** [comment]
I'm running v5.3.7 originally installed via git.  When I run "php upgrade.php" it seemingly goes through the motions but leaves me on V5.3.7.  What am I missing?

**Reply by adagioajanes** [reply]
@iamkhris the `php upgrade.php` process defaults to the latest master release. This is still a release candidate, so it is not captured by that upgrade process by default.

I recommend making a copy of your production environment while testing this as well, rather than upgrading production just yet. This is still a release candidate, so there may be a few quality items to address first.

**Reply by iamkhris** [reply]
Gotcha.  Thank you. :)

**Reply by snipe** [reply]
Once you have a production copy that you can mess up while testing, you can use `php upgrade.php develop` and it will pull the latest from the `develop` branch, which is what the RC is based off of.

**Comment by cswinton147** [comment]
great work on getting a new release out (later than expected or not) during a pandemic. can't wait to try this.

**Comment by bunzladi** [comment]
If you working with PHP 7.4.13 you must activate the PHP Sodium Extension.
You have to go to the PHP.ini file and at the very bottom add the "extension=php_sodium.dll" to the list. Otherwise your Snipe IT will be down and you will not be able to downgrade or upgrade it.
But otherwise everything worked well.

**Reply by rfrost** [reply]
**Oracle Linux Users**

If you are running Oracle Linux 7.x ( or 8 )  with Oracle's mirrored EPEL you cannot install the Sodium library because the php-pecl-sodium RPM is out of date and dependent on Version 5 of php. So the workaround is to compile your own PECL sodium library and the steps are as follows:

NOTE:  sodium.so file for PHP needs to be compiled with PECL. But this process requires EPEL.
NOTE:  Sodium is from EPEL. Enable Oracle's Mirrored EPEL, install build tools needed for compiling extensions

```
yum -y install epel-release
yum -y groupinstall "Development Tools"
yum -y install libsodium libsodium-devel
pecl install libsodium
```

Then add the  "extension=sodium.dll/sodium.so" to the PHP.ini file

Check that the sodium module is loaded by searching for it by running "php -m" to enumerate the PHP module list

Then proceed to do the normal php upgrade
